src/StmtToViz.cpp

```diff
@@ -0,0 +1,2900 @@
+    }
+
+    void visit(const LT *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const LE *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const GT *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const GE *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const And *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const Or *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get(), op->b.get()});
+        set_data_costs(op, 0, {op->a.get(), op->b.get()});
+    }
+
+    void visit(const Not *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->a.get()});
+        set_data_costs(op, 0, {op->a.get()});
+    }
+
+    void visit(const Select *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, op->type.lanes(), {op->condition.get(), op->true_value.get(), op->false_value.get()});
+        set_data_costs(op, 0, {op->condition.get(), op->true_value.get(), op->false_value.get()});
+    }
+
+    void visit(const Load *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->predicate.get(), op->index.get()});
+        set_data_costs(op, op->type.bits() * op->type.lanes(), {op->predicate.get(), op->index.get()});
+    }
+
+    void visit(const Ramp *op) override {
+        // The cost of a Ramp is higher when the stride is not 1,
+        // but currently the cost model does not consider such cases
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->base.get(), op->stride.get()});
+        set_data_costs(op, 0, {op->base.get(), op->stride.get()});
+    }
+
+    void visit(const Broadcast *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 1, {op->value.get()});
+        set_data_costs(op, 0, {op->value.get()});
+    }
+
+    void visit(const Call *op) override {
+        IRVisitor::visit(op);
+        std::vector<const IRNode *> args;
+        for (const auto &arg : op->args) {
+            args.push_back(arg.get());
+        }
+        set_compute_costs(op, 1, args);
+        // Currently there is no special handling
+        // for intrinsics such as `prefetch`
```

Comment:
Do we think they will need special handling? Does this require a TODO or a tracking issue?

Reply:
I'm unsure what the best course of action is here. A prefetch performs data-movement that is important to highlight (esp for backends like HVX). Otoh, unless the prefetch parameters are constants, it makes the estimated cost a symbolic expression. I worry that updating the cost model and visualizer code to handle symbolic expressions sensibly will grow the code by a non-trivial amount for perhaps very little gain. We could assign some constant cost to every prefetch to ensure it gets highlighted but that would not be true to the description of the cost model (counting bits moved).